core/src/main/java/org/apache/iceberg/rest/responses/LoadTableResponse.java

```diff
@@ -106,6 +113,8 @@ public Builder addAllConfig(Map<String, String> properties) {
 
     public LoadTableResponse build() {
       Preconditions.checkNotNull(metadata, "Invalid metadata: null");
+      // This is causing errors in TestRESTCatalog via CatalogHandlers
+      Preconditions.checkNotNull(metadataLocation, "Invalid metadata location: null");
```

Comment:
This is the check that is causing 16 test failures in `TestRESTCatalog` - which has been reduced to 13 by the changes in `CatalogHandlers`.

Without this check, all tests in `TestRESTCatalog` pass. See notes here too